query/src/test/java/org/infinispan/query/blackbox/LocalCacheTest.java

```diff
@@ -344,53 +318,44 @@ public void testIteratorRemove() {
 
    public void testLazyIteratorWithOffset() {
       loadTestingData();
-      CacheQuery<?> cacheQuery = createQuery("blurb:'Eats'", Person.class).firstResult(1);
+      Query<?> cacheQuery = createQuery("blurb:'Eats'", Person.class).startOffset(1);
 
-      try (ResultIterator<?> iterator = cacheQuery.iterator(new FetchOptions().fetchMode(FetchOptions.FetchMode.LAZY))) {
+      try (CloseableIterator<?> iterator = cacheQuery.iterator()) {
          assertEquals(1, countElements(iterator));
       }
       StaticTestingErrorHandler.assertAllGood(cache);
    }
 
    @Test(expectedExceptions = IllegalArgumentException.class)
    public void testSearchManagerWithNullCache() {
-      Search.getSearchManager(null);
+      Search.getQueryFactory(null);
    }
 
-   @Test(expectedExceptions = IllegalArgumentException.class)
-   public void testLazyIteratorWithInvalidFetchSize() {
-      loadTestingData();
-      CacheQuery<?> cacheQuery = createQuery("blurb:'Eats'", Person.class).firstResult(1);
-
-      cacheQuery.iterator(new FetchOptions().fetchMode(FetchOptions.FetchMode.LAZY).fetchSize(0));
-   }
+// TODO: Allow to specify the fetch size
```

Comment:
Do we have a jira for this @gustavonalle ?

Reply:
https://issues.redhat.com/browse/ISPN-11881